tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -3853,13 +3863,23 @@ SpirvInstruction *SpirvEmitter::processRWByteAddressBufferAtomicMethods(
     }
   } else {
     auto *value = doExpr(expr->getArg(1));
+
+    spv::Op atomicOp = translateAtomicHlslOpcodeToSpirvOpcode(opcode);
+    if (atomicOp == spv::Op::OpAtomicUMax && baseType->isSignedIntegerType())
+      atomicOp = spv::Op::OpAtomicSMax;
+    if (atomicOp == spv::Op::OpAtomicSMax && baseType->isUnsignedIntegerType())
+      atomicOp = spv::Op::OpAtomicUMax;
+    if (atomicOp == spv::Op::OpAtomicUMin && baseType->isSignedIntegerType())
+      atomicOp = spv::Op::OpAtomicSMin;
+    if (atomicOp == spv::Op::OpAtomicSMin && baseType->isUnsignedIntegerType())
+      atomicOp = spv::Op::OpAtomicUMin;
```

Comment:
These changes are not needed. I believe this will give incorrect results. The parser already determine the correct opcode.